libs/api/domains/cms/src/lib/search/importers/organizationSubpage.service.ts

```diff
@@ -0,0 +1,56 @@
+import { MappedData } from '@island.is/content-search-indexer/types'
+import { logger } from '@island.is/logging'
+import { Injectable } from '@nestjs/common'
+import { Entry } from 'contentful'
+import { IOrganizationSubpage } from '../../generated/contentfulTypes'
+import { CmsSyncProvider, processSyncDataInput } from '../cmsSync.service'
+import { mapOrganizationSubpage } from '../../models/organizationSubpage.model'
+
+@Injectable()
+export class OrganizationSubpageSyncService
+  implements CmsSyncProvider<IOrganizationSubpage> {
+  processSyncData(entries: processSyncDataInput<IOrganizationSubpage>) {
+    return entries.filter(
+      (entry: Entry<any>): entry is IOrganizationSubpage =>
+        entry.sys.contentType.sys.id === 'organizationSubpage',
```

Comment:
We probably want to check if `title` field is set here else elasticsearch will import empty pages for all set locales.
We consider empty title fields as empty documents this is not a problem for Icelandic as field validation is respected when publishing the main locale.

You might also want to check if the entry is a circular structure and take appropriate action such as pruning the data.
This can happen if this entry has related entries as reference and one of the related entries has this entry as reference or in other similar cases.
A circular structure would throw an error in `JSON.stringify` this would cause the entry to not be imported and would log a warning in the importer.

This applies to all importers

Reply:
I've added null check for the title fields. I think circular checks are unnecessary because the Contentful schema for these types doesn't have circles.